src/components/Configure/layout/ConditionalProxyLayout/ConditionalProxyLayout.tsx

```diff
@@ -1,39 +1,68 @@
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
 
 import { LoadingIcon } from '../../../../assets/LoadingIcon';
+import { useApiKey } from '../../../../context/ApiKeyContextProvider';
+import { useConnections } from '../../../../context/ConnectionsContextProvider';
+import { useInstallIntegrationProps } from '../../../../context/InstallIntegrationContextProvider';
+import { useProject } from '../../../../context/ProjectContextProvider';
+import { hasKeys } from '../../../../utils';
+import { ErrorTextBox } from '../../../ErrorTextBox';
+import { createInstallationProxyOnly } from '../../actions/proxy/createInstallationProxyOnly';
 import { useHydratedRevision } from '../../state/HydratedRevisionContext';
 
+import { nonProxyActionList } from './constants';
 import { InstalledSuccessBox } from './InstalledSuccessBox';
 
 interface ConditionalProxyLayoutProps {
   children: React.ReactNode;
 }
 
 /**
- * if the hydratedRevision does not have read or write,
+ * if the hydratedRevision only has proxy actions,
  * then it will not render the ConfigureInstallation
  * @returns
  */
 export function ConditionalProxyLayout({ children }: ConditionalProxyLayoutProps) {
-  const { hydratedRevision, loading } = useHydratedRevision();
-  const provider = hydratedRevision?.content?.provider;
+  const { projectId } = useProject();
+  const apiKey = useApiKey();
+  const { hydratedRevision, loading: hydratedRevisionLoading } = useHydratedRevision();
+  const {
+    integrationObj, installation, groupRef, consumerRef, setInstallation,
+  } = useInstallIntegrationProps();
+  const { selectedConnection } = useConnections();
+  const [createInstallLoading, setCreateInstallLoading] = useState(false);
+  const isLoading = hydratedRevisionLoading || createInstallLoading;
 
-  // // TMP remove read and write
-  // if (hydratedRevision) {
-  //   hydratedRevision.content.read = undefined;
-  //   hydratedRevision.content.write = undefined;
-  // }
+  const provider = hydratedRevision?.content?.provider;
 
-  const isProxy = !hydratedRevision?.content.read && !hydratedRevision?.content.write;
+  const actionKeys = Object.keys(hydratedRevision?.content || {});
+  const isProxyOnly: boolean = hasKeys(actionKeys, nonProxyActionList);
```

Comment:
To clarify, my last comment wasn't about the naming, it's about the logic. This should be:

```
const isProxyOnly: boolean = !hasKeys(actionKeys, nonProxyActionList);
```

Reply:
could catch, thankfully this also helped find the other change with read & write keys.